cmd/thanos/receive.go

```diff
@@ -859,6 +861,8 @@ func (rc *receiveConfig) registerFlag(cmd extkingpin.FlagClause) {
 
 	rc.tsdbMaxBlockDuration = extkingpin.ModelDuration(cmd.Flag("tsdb.max-block-duration", "Max duration for local TSDB blocks").Default("2h").Hidden())
 
+	rc.tsdbOutOfOrderTimeWindow = extkingpin.ModelDuration(cmd.Flag("tsdb.out-of-order-time-window", "[EXPERIMENTAL] Configures the allowed time window for ingestion of out-of-order samples. Disabled (0s) by default").Default("0s").Hidden())
```

Comment:
With OOO samples enabled, we will create overlapping time range blocks so vertical compaction needs to be enabled at the compactor side. Since https://github.com/thanos-io/thanos/pull/5794 is still ongoing shall we mention to enable this flag?

Reply:
Is this still true after vertical compaction is enabled by default on Prometheus (https://github.com/prometheus/prometheus/issues/11255)?

Reply:
On the TSDB side we should be side. But since we have additional overlapping checks on the compactor side then we will fail...

Reply:
Got it, I added a small disclaimer :+1: